nautobot/core/tests/integration/test_view_authentication.py

```diff
@@ -0,0 +1,71 @@
+from django.test import tag
+
+from nautobot.core.testing import TestCase
+from nautobot.core.utils.lookup import get_url_for_url_pattern, get_url_patterns
+
+
+@tag("integration")
+class AuthenticationEnforcedTestCase(TestCase):
+    r"""
+    Test that all\* registered views require authentication to access.
+
+    \* with a very small number of known exceptions such as login and logout views.
+    """
+
+    def test_all_views_require_authentication(self):
+        self.client.logout()
+        url_patterns = get_url_patterns()
+
+        for url_pattern in url_patterns:
+            with self.subTest(url_pattern=url_pattern):
+                url = get_url_for_url_pattern(url_pattern)
+                response = self.client.get(url, follow=True)
+
+                if response.status_code == 405:  # Method not allowed
+                    response = self.client.post(url, follow=True)
+
+                # Is a view that *should* be open to unauthenticated users?
+                if url in [
+                    "/admin/login/",
+                    "/api/plugins/example-plugin/webhook/",
+                    "/api/redoc/",
+                    "/api/swagger/",
+                    "/api/swagger.json",
+                    "/api/swagger.yaml",
```

Comment:
This is at the very least inconsistent - we don't allow access to `/api/docs/` for unauthenticated users, but we do permit `/api/redoc/`? And we let unauthenticated users download the schema? I think we should probably block unauthenticated access to those endpoints - thoughts?

Reply:
I feel like disabling all endpoints mentioned above for unauthenticated users should be the way to go. Or could we allow `/api/docs` for unauthenticated users? Is there any sensitive information on that endpoint?

Reply:
We made `/api/docs/` require authentication in #4068 and #4865. I think we should add the restriction to these endpoints rather than removing it from `/api/docs/`

Reply:
Agreed this is the plan.